asynch/pool.py

```diff
@@ -157,22 +176,43 @@ async def _release_connection(self, conn: Connection) -> None:
             raise AsynchPoolError(f"the connection {conn} does not belong to {self}")
 
         self._acquired_connections.remove(conn)
+        try:
+            await conn._refresh()
+        except ConnectionError as e:
+            msg = f"the {conn} is invalidated"
+            raise AsynchPoolError(msg) from e
+
         self._free_connections.append(conn)
 
-    async def _init_connections(self, n: Optional[int] = None) -> None:
-        to_create = n if n is not None else self.minsize
-        if to_create < 0:
-            msg = f"cannot create ({to_create}) negative connections for {self}"
+    async def _init_connections(self, n: int, *, strict: bool = False) -> None:
+        if n < 0:
+            msg = f"cannot create a negative number ({n}) of connections for {self}"
             raise ValueError(msg)
-        if to_create == 0:
-            return
-        if (self.connections + to_create) > self.maxsize:
-            msg = f"cannot create {to_create} connections to exceed the size of {self}"
+        if (self.pool_size + n) > self.maxsize:
+            msg = (
+                f"{self} has the {self.pool_size} connections, "
+                f"adding {n} will exceed its maxsize ({self.maxsize})"
+            )
             raise AsynchPoolError(msg)
+        if not n:
+            return
+
+        # it is possible that the `_create_connection` may not create `n` connections
         tasks: list[asyncio.Task] = [
-            asyncio.create_task(self._create_connection()) for _ in range(to_create)
+            asyncio.create_task(self._create_connection()) for _ in range(n)
         ]
-        await asyncio.wait(fs=tasks)
+        # that is why possible exceptions from the `_create_connection` are also gathered
+        if strict and any(
+            i
+            for i in await asyncio.gather(*tasks, return_exceptions=True)
+            if isinstance(i, Exception)
+        ):
```

Comment:
Is it more readable to unwind this?
```py
if strict:
    results = await asyncio.gather(...)
    for result in results:
        if isinstance(result, Exception):
            raise ...
```

Reply:
I believe it is readable enough, the cognitive load is not hefty, the `any` builtin eliminates the necessity of `for` and `if` constructs. I addition, no need in any intermediate variable.

Reply:
It is up to you, feel free to resolve this